src/averageAges.js

```diff
@@ -55,11 +84,26 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const children = people.filter(({ mother, sex }) => (
+    people.some(({ name }) => name === mother) && (
+      onlyWithSon
+        ? sex === MAN
+        : true
+    )
+  ));
+
+  const differenceOfAge = children.reduce(
+    (acc, { mother, born: childBorn }) => {
+      const { born: motherBorn } = people.find(({ name }) => mother === name);
+
+      return acc + (childBorn - motherBorn);
+    }, COUNT);
+
+  return differenceOfAge / children.length;
 }
 
 module.exports = {
   calculateMenAverageAge,
   calculateWomenAverageAge,
   calculateAverageAgeDiff,
-};
+};
```

Comment:
Add [empty line in the end of file](https://unix.stackexchange.com/questions/18743/whats-the-point-in-adding-a-new-line-to-the-end-of-a-file)